### PR TITLE
config: remove txn-local-latches from config file

### PR DESCRIPTION
**Pull request by bobotu**
### What problem does this PR solve?
The `txn-local-latches` may lack sufficient tests and not recommend to use for now.

### What is changed and how it works?
Omitting `txn-local-latches` when marshal/unmarshals config file and make it disabled by default.

Tests 
- Unit test

### CODE REVIEW

**Comment by bobotu** [comment]
PTAL @jackysp

File: config/config.go [hunk 1]
```diff
@@ -77,7 +77,7 @@ type Config struct {
 	MemQuotaQuery    int64           `toml:"mem-quota-query" json:"mem-quota-query"`
 	EnableStreaming  bool            `toml:"enable-streaming" json:"enable-streaming"`
 	EnableBatchDML   bool            `toml:"enable-batch-dml" json:"enable-batch-dml"`
-	TxnLocalLatches  TxnLocalLatches `toml:"txn-local-latches" json:"txn-local-latches"`
+	TxnLocalLatches  TxnLocalLatches `toml:"-" json:"-"`
```
Comment (jackysp) [inline config/config.go]: Will it make `-config-check` failed? Maybe we need to add it to https://github.com/pingcap/tidb/blob/7a67ff482b21b7405b582d799215f5ac333fc0ec/config/config.go#L675 .

**Comment by tiancaiamao** [comment]
LGTM

**Pull request merged**